server/aesdsocket.c

```diff
@@ -0,0 +1,323 @@
+	}
+	//After creation of the socket, bind function binds the socket to address and port number
+	status = bind(sockfd, serveinfo->ai_addr,  serveinfo->ai_addrlen);
+	if(status == -1 )        
+	{
+		syslog(LOG_ERR,"bind() failed\n\r");
+		printf("bind error: %s\n",strerror(errno));
+		freeaddrinfo(serveinfo);
+		return -1;
+	}
+	
+	//Daemonize the process after bind is successfull
+	if(argc >= 2 && argv[1] != NULL && (strcmp(argv[1],"-d") == 0))
+	{
+	/* Reference: https://stackoverflow.com/questions/2966886/is-there-a-difference-calling-daemon0-0-from-within-a-program-and-launching-a
+    	Re-parents the process to init by forking and then exiting the parent. Look in the ps list and you'll see that daemons are owned by PID 1.
+    	Calls setsid().
+    	Changes directory to /.
+    	Redirects standard in, out, and error to /dev/null.
+	*/
+
+		status = daemon(0,0);
+		if(status == -1)
+		{
+			syslog(LOG_ERR,"daemon failed");
+			return -1;
+		}
+	}
+
+	//Listen
+	status = listen(sockfd,MAX_PENDING_CONN_REQUEST) ;
+	if(status == -1)
+	{
+		syslog(LOG_ERR,"listen() failed\n\r");
+		freeaddrinfo(serveinfo);
+		return -1;
+	}
+
+	//open file to write the received data from client
+	FILE* fptr = fopen(RECV_FILE_NAME,"w"); //use a+ to open already existing file, w to create new file if not exist 
+	if(fptr == NULL)
+	{
+		syslog(LOG_ERR,"fopen() \n\r");
+		freeaddrinfo(serveinfo);
+		return -1;	
+	}
+
+	fclose(fptr);
+	fptr = NULL;
+
+	while(1)
+	{
+		syslog(LOG_ERR,"Waiting for connection from client() \n\r");
+		//Accept
+		//struct sockaddr client_addr; 
+		memset(&client_addr, 0, sizeof(client_addr));
+		socklen_t client_addr_len = 0;
+
+
+		accepted_sockfd = accept(sockfd, (struct sockaddr*)&client_addr, &client_addr_len);	
+
+		if(accepted_sockfd == -1)
+		{
+			syslog(LOG_ERR,"accept() \n\r");
+			freeaddrinfo(serveinfo);
+			return -1;
+		}	
+
+
+		//syslog(LOG_ERR,"Accepted connection from %d%d%d%d\n\r",);
+		//printf("Client Adress = %s",inet_ntoa(client_addr.sin_addr));
+		syslog(LOG_ERR,"Accepted connection from %s\n\r",inet_ntoa(client_addr.sin_addr)); //inet_ntoa converts raw address into human readable format
+
+
+		//open file to write the received data from client
+		fptr = fopen(RECV_FILE_NAME,"a"); //use a+ to open already existing file, w to create new file if not exist 
+		if(fptr == NULL)
+		{
+			syslog(LOG_ERR,"fopen() \n\r");
+			freeaddrinfo(serveinfo);
+			return -1;	
+		}
+
+		//Receive data from client
+		const int recv_len = 100;
+		// char recv_data[RECV_LEN];
+		recv_data = malloc(sizeof(char) * recv_len);
+		memset(recv_data,0,recv_len*sizeof(char));
+
+    	int n = 1, total = 0, found = 0;
+		while (!found) 
+		{
+			n = recv(accepted_sockfd, &recv_data[total], recv_len, 0);
+			if (n == -1) 
+			{
+				syslog(LOG_ERR,"\n\rrecv failed ");
+				freeaddrinfo(serveinfo);
+				return -1;	
+			}
+			total += n;
+			found = (recv_data[total - 1] == '\n')?(1):(0);
+			recv_data = realloc(recv_data, total + recv_len);
```

Comment:
@samu7988 what assumption is made here about the size of RAM heap relative to the total size of all packets sent?